htlcswitch/link.go

```diff
@@ -1970,6 +1970,23 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 			l.channel.RevokeCurrentCommitment()
 		if err != nil {
 			l.log.Errorf("unable to revoke commitment: %v", err)
+
+			// We need to fail the channel in case revoking our
```

Comment:
Let's also do this for any of the other related channel operations? So process revoke, send sig, etc.

Reply:
Thanks for this headup,

After going through the code it seems to me that we are failing the channel accordingly already for most of the situations?

- After processing(receiving) the revocation msg: https://github.com/lightningnetwork/lnd/blob/master/htlcswitch/link.go#L2037
- After receiving feeupdate msg: https://github.com/lightningnetwork/lnd/blob/master/htlcswitch/link.go#L2107
- Also when receiving a commitment: https://github.com/lightningnetwork/lnd/blob/master/htlcswitch/link.go#L1940-L1961
- When signing a new commitment we fail the channel if an error occurs: https://github.com/lightningnetwork/lnd/blob/master/htlcswitch/link.go#L2206

So I think this was the only gap we had not failing the channel. But maybe you were referring to something different ?

But I will look again to see whether we could still have a case where we do not fail. I will update on this issue.

Reply:
I am currently thinking about returning a warning in situations where we wanna give the channel time to resolve its issue (not an internal error basically) wdyt?

Reply:
> I am currently thinking about returning a warning in situations where we wanna give the channel time to resolve its issue (not an internal error basically) wdyt?

Thinking through the warning idea again, I think this might not come in handy if we have several channels per peer, and according to the spec we would also need to close the peer connection, therefore affecting also other channels of the peer.

Reply:
keeping it with the internal error for now

Reply:
Yeah what's important here though is that we teat down the connection to ensure we don't retain any unclean state.

Reply:
> n, I think this might not come in handy if we have several channels per peer, and according to the spec we would also need to close the peer connection, therefore affecting also other channels of the peer.

So I ended up going in this direction, as otherwise this could happen:
  * DB messes up 
  * We just send the error/warning 
  * Peer doesn't try to tear down the link for w/e reason 
  * They send an add+sig
  * We never get this (no link up), eventual force close occurs 

Just have it as a single commit on top, we can remove otherwise though.